### PR TITLE
Add a tutorial with OpenAI function calling

### PR DESCRIPTION
**Pull request by bilgeyucel**

### CODE REVIEW

**Comment by bilgeyucel** [comment]
@vblagoje, I’d appreciate it if you could provide your feedback in terms of technical terms and the use case

File: tutorials/40_Building_Agent-like_Application_with_Function_Calling.ipynb [hunk 1]
```diff
@@ -0,0 +1,6329 @@
+{
```
Comment (TuanaCelik) [inline tutorials/40_Building_Agent-like_Application_with_Function_Calling.ipynb:1]: <p>Agent isn't an application imo, you build an application <em>with</em> an agent. So maybe "Function Calling for Agent-like Behaviour/Functionality"?? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepset-ai/haystack-tutorials/pull/301/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

File: index.toml [hunk 2]
```diff
@@ -396,3 +396,14 @@ aliases = []
 completion_time = "10 min"
 created_at = 2024-02-20
 haystack_2 = true
+
+[[tutorial]]
+title = "Building an Agent-like Application with Function Calling"
+description = "Learn how to use OpenAIChatGenerator to build chat applications using tools"
+level = "intermediate"
```
Comment (TuanaCelik) [inline index.toml:403]: advanced? Wdyt? I am open to it.. :)

File: tutorials/40_Building_Agent-like_Application_with_Function_Calling.ipynb (hunk 1 above)
Comment (vblagoje) [reply tutorials/40_Building_Agent-like_Application_with_Function_Calling.ipynb:1]: Yeah, I agree with @TuanaCelik here

Comment (bilgeyucel) [reply tutorials/40_Building_Agent-like_Application_with_Function_Calling.ipynb]: changed the name 👍

**Pull request merged**